### ISSUE TITLE
Allow to bench custom network topologies

### ISSUE THREAD

**Issue by fyrchik** [nspcc-dev/neo-bench#92]
In real networks topology is far from full mesh. It would be nice if we could bench such topologies too.
The easiest way to achieve this is to restrict `MaxPeers` to 2, but we also need to be sure that each node has it's way to RPC node.
This can be achieved either with some custom docker network configuration or by explicitly blacklisting IPs inside the container.

**Comment by roman-khimov** [comment]
I don't think it makes much sense for a network with 4 CNs, but I'd really like to see 7 CNs plus 2 RPC nodes setup with a limit of up to 3-4 connections per node, that'd be interesting. Also, we probably want to have some well-known set of topologies to compare different results against each other.

**Comment by fyrchik** [comment]
@roman-khimov Actually I already have some interesting results even for 4 CNs with line/star topologies, where one of the nodes tries to initiate change view and/or receives new blocks before the commit messages for them.

The problem with just restricting `MaxPeers` is less control over the actual topology and thus less deterministic results (e.g. for 7 nodes we can have a single bridge between 2 almost full-mesh clusters or a more balanced variant. Also, the load depends on which node will be connected with the RPC, we can even fail to run bench if the first one is not).

I think the best solution is:
1. Generate subnet for each connected set of nodes (e.g. to connect 4 nodes in line we have `neo_go_12`, `neo_go_23`, `neo_go_34` networks with each node except the last belonging to 2 networks + the first one uses `default` to connect with RPC node).
2. Have single docker-compose template (this also helps to unify go/mixed/sharp as well as 7+2 setups).
3. The actual files can be written by hand (or with some generation utility) or generated on the fly based on simple descriptions (star.4, mesh.7 etc.).

(1) also allows to perform traffic shaping on a per-subnet basis (subnets should correspond to different interfaces in containers, though I haven't checked this).
We can do (2) separately in #95 .
